### PR TITLE
Add free threaded Python 3.14t support to build workflow

### PR DESCRIPTION
**Pull request by cclauss**
% `uvx --with=lz4 python3.14t -c"import lz4"`
```
<frozen importlib._bootstrap>:491: RuntimeWarning: The global interpreter lock (GIL) has been enabled
    to load module 'lz4._version', which has not declared that it can run safely without the GIL.
    To override this behavior and keep the GIL disabled (at your own risk), run with PYTHON_GIL=0
    or -Xgil=0.
```
These are expected to fail.  ___Help needed___ to fix the failing free threaded builds.
https://py-free-threading.github.io/porting

Related:
* #304

### CODE REVIEW

**Comment by ngoldbaum** [comment]
I think the failing test probably indicates that the memory usage pattern is different on 3.14t and the test failure is probably "real" in the sense that the upper bound is being violated for whatever reason. There were some [pretty substantial changes to the free-threaded GC in 3.14t](https://labs.quansight.org/blog/free-threaded-gc-3-14), maybe we're hitting a corner case? I'll see if Neil wants to take a closer look at this...

@cclauss just because a test is marked as thread-unsafe doesn't mean that the test is *expected* to fail on the free-threaded build. Only that it is an inherently thread-unsafe test and should be run on one core when running under pytest-run-parallel. You can also tell pytest-run-parallel to skip thread-unsafe tests, but that's not the default.

@ogrisel one thing that might help future readers to interpret the thread-unsafe markers is to also specify a reason. In this case, it would have something like:

```
@pytest.mark.thread_unsafe(reason="Test uses tracemalloc to estimate global memory usage")
```

The reason will also get propagated to the verbose output, allowing people to see why an unsafe test is skipped.

We should probably make pytest-run-parallel warn when people use the thread-unsafe marker without a reason. I tried to figure out how to do that at one point but got stuck and haven't come back to it.

File: tests/block/test_block_3.py [hunk 1]
```diff
@@ -1,8 +1,14 @@
 import gc
+import os
```
Comment (ogrisel) [inline tests/block/test_block_3.py]:
```suggestion
```

**Review by ogrisel — APPROVED** [review]
LGTM as it is. The test failure on ubuntu and windows with free-threading might be revealing a real problem w.r.t. memory usage expectations, but it's indeed hard to tell, as @ngoldbaum explained above. I could not trigger the problem locally on macOS to investigate myself.

However, the core functionality of lz4 seems to work as expected on cp314t, so I would not condition the merge of this PR to the results of the investigation of the higher than expected memory usage reported by `tracemalloc` in this test.

Having cp314t wheels for python-lz4 would make it easier for downstream projects to get their own CI running on cp314t and subsequently unlock free-threading adoption throughout the Python ecosystem.
